src/overlays/actors/ovl_Bg_Ydan_Hasi/z_bg_ydan_hasi.c

```diff
@@ -21,23 +27,180 @@ const ActorInit Bg_Ydan_Hasi_InitVars = {
+    this->dyna.actor.posRot.pos.x =
+        ((Math_Sins(this->dyna.actor.posRot.rot.y) * framesAfterMath) + this->dyna.actor.initPosRot.pos.x);
+    this->dyna.actor.posRot.pos.z =
+        ((Math_Coss(this->dyna.actor.posRot.rot.y) * framesAfterMath) + this->dyna.actor.initPosRot.pos.z);
+    waterBox = &globalCtx->colCtx.stat.colHeader->waterBoxes[1];
+    this->dyna.actor.posRot.pos.y = waterBox->unk_02 + 20.0f;
+    if (this->timer != 0) {
+        this->timer--;
+    }
+    if (this->timer == 0) {
+        this->timer = 50;
+    }
+    posOffset = sinf(this->timer * 0.12566371f);
+    this->dyna.actor.posRot.pos.y = (this->dyna.actor.posRot.pos.y + (posOffset + posOffset));
+}
+
+void BgYdanHasi_InitWater(BgYdanHasi* this, GlobalContext* globalCtx) {
+    if (Flags_GetSwitch(globalCtx, (u8)this->unk_168)) {
+        this->timer = 600;
+        this->actionFunc = BgYdanHasi_MoveWater;
+    }
+}
+
+WaterBox* BgYdanHasi_MoveWater(BgYdanHasi* this, GlobalContext* globalCtx) {
+    WaterBox* waterBox;
+
+    if (this->timer == 0) {
+        if (Math_ApproxF(&this->dyna.actor.posRot.pos.y, this->dyna.actor.initPosRot.pos.y, 1.0f) != 0) {
+            Flags_UnsetSwitch(globalCtx, this->unk_168);
+            this->actionFunc = BgYdanHasi_InitWater;
+        }
+        func_8002F948(&this->dyna.actor, NA_SE_EV_WATER_LEVEL_DOWN - SFX_FLAG);
+    } else {
+        if (Math_ApproxF(&this->dyna.actor.posRot.pos.y, this->dyna.actor.initPosRot.pos.y - 47.0f, 0.5f)) {
+            this->actionFunc = BgYdanHasi_DecWaterTimer;
+        }
+        func_8002F948(&this->dyna.actor, NA_SE_EV_WATER_LEVEL_DOWN - SFX_FLAG);
+    }
+    waterBox = globalCtx->colCtx.stat.colHeader->waterBoxes;
+    globalCtx->colCtx.stat.colHeader->waterBoxes[1].unk_02 = this->dyna.actor.posRot.pos.y;
+    if (1) {}
+    return waterBox + 0x1;
+}
+
+void BgYdanHasi_DecWaterTimer(BgYdanHasi* this, GlobalContext* globalCtx) {
+    if (this->timer != 0) {
+        this->timer--;
+    }
+    func_8002F994(&this->dyna.actor, this->timer);
+    if (this->timer == 0) {
+        this->actionFunc = BgYdanHasi_MoveWater;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ydan_Hasi/func_808BE690.s")
+void BgYdanHasi_SetupThreeBlocks(BgYdanHasi* this, GlobalContext* globalCtx) {
+    if (Flags_GetSwitch(globalCtx, this->unk_168) != 0) {
+        this->timer = 260;
+        this->dyna.actor.draw = BgYdanHasi_Draw;
+        this->actionFunc = BgYdanHasi_UpdateThreeBlocks;
+        func_800800F8(globalCtx, 0xBE0, 0x1E, &this->dyna.actor, 0);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ydan_Hasi/func_808BE7C8.s")
+void BgYdanHasi_UpdateThreeBlocks(BgYdanHasi* this, GlobalContext* globalCtx) {
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ydan_Hasi/func_808BE810.s")
+    if (this->timer != 0) {
+        this->timer--;
+    }
+    if (this->timer == 0) {
```

Comment:
I think the returns here can be removed and turn these blocks into if-elses

Reply:
I tried this before and it wasn’t working